code.gs

```diff
@@ -400,14 +415,32 @@ MergedContact.prototype.getInfoFromContact = function (contactId) {
     null                          // Profile image URL.
   ));
 
+  // Events blacklist.
+  blacklist = googleContact.getCustomFields('notificationBlacklist');
+  if (blacklist && blacklist[0]) {
+    self.blacklist = uniqueStrings(blacklist[0].getValue().replace(/,+/g, ',').replace(/(^,|,$)/g, '').split(','));
+  }
+
   // Events.
   googleContact.getDates().forEach(function (dateField) {
-    self.addToField('events', new EventDC(
-      String(dateField.getLabel()),
-      dateField.getYear(),
-      monthToInt(dateField.getMonth()) + 1,
-      dateField.getDay()
-    ));
+    var thisEventLabel, thisEventMonth, thisEventDay;
+
+    thisEventLabel = String(dateField.getLabel());
+    if (thisEventLabel !== 'BIRTHDAY' && thisEventLabel !== 'ANNIVERSARY') {
+      thisEventLabel = 'CUSTOM:' + thisEventLabel;
+    }
+    thisEventMonth = monthToInt(dateField.getMonth()) + 1;
+    thisEventDay = dateField.getDay();
+    if (((!eventDay && !eventMonth) ||
+         (thisEventMonth === eventMonth && thisEventDay === eventDay)) &&
+        (!self.blacklist || !self.blacklist.length || !isIn(beautifyLabel(thisEventLabel), self.blacklist))) {
```

Comment:
I think this conditional should be formatted differently: as it is now it's quite difficult to grasp operators precedences and meanings.

It could use some more spacing (not so pleasing to look at, but easy to understand at first glance):

```
if (
  (
    (!eventDay && !eventMonth) ||
    (thisEventMonth === eventMonth && thisEventDay === eventDay)
  ) && (
    !self.blacklist || !self.blacklist.length || !isIn(beautifyLabel(thisEventLabel), self.blacklist)
  )
) {
```

or extracting partial conditions (prettier in my opinion, but still easily understandable):

```
dateMatches = (!eventDay && !eventMonth) || (thisEventMonth === eventMonth && thisEventDay === eventDay));
blacklistMatches = self.blacklist && self.blacklist.length && isIn(beautifyLabel(thisEventLabel), self.blacklist)
if (dateMatches && !blacklistMatches) {
```

Reply:
OK, in a while I will change it to use the partially extracted conditions you showed (or if you prefer not to wait feel free to add that change as a commit to the PR branch).